inc/Symbol.h

```diff
@@ -0,0 +1,81 @@
+////////////////////////////////////////////////////////////////////////////////
+/**
+ * @file Symbol.h
+ * @date Wed February 23 2022
+ * @brief Header for a class which can be used to represent a symbol in a 
+ * mathematical expression and, optionally, its associated numerical value.
+ */
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef SYMBOL_H
+#define SYMBOL_H
+
+#include <iostream>
+
+namespace calculator {
+
+/**
+ * @brief Represents a symbol in a mathematical expression. Symbols have a 
+ * single associated glyph from a-z and can optionally have a value (e.g. int or
+ * double) bound to them.
+ */
+template<typename T>
+class Symbol
+{
+    public:
+        /**
+         * @brief Default value returned by getValue() when no value has been
+         * bound to the symbol
+         * @return T The default value
+         */
+        static T defaultValue();
+
+        /**
+         * @brief Construct a new Symbol object
+         * @param symbol the character glyph associated with the object
```

Comment:
for parameters in functions, it should look like this
@param [in] symbol ....
or
@param [out] ....
The in is used for inputs parameters and out for output parameters.  So do that for all your functions that take parameters